ex2_hal/adcs/equipment_handler/src/adcs_io.c

```diff
@@ -203,36 +196,55 @@ ADCS_returnState request_uart_telemetry(uint8_t TM_ID, uint8_t *telemetry, uint3
  *    the actual image data
  *
  */
-ADCS_returnState receive_uart_packet(uint8_t *hole_map, uint8_t *image_bytes) {
+ADCS_returnState receive_file_download_uart_packet(uint8_t *pckt, uint16_t *pckt_counter) {
+    if (xSemaphoreTake(uart_mutex, UART_TIMEOUT_MS) != pdTRUE) {
+        return ADCS_UART_FAILED;
```

Comment:
consider using an ADCS_UART_BUSY to distinguish between communication failure and busy states